### PR TITLE
saas 11.2 im_support test suite

### PR DESCRIPTION
**Pull request by aab-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/im_support/static/tests/systray_no_support_tests.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+QUnit.module('systray', {
+    beforeEach: function () {
+        this.services = [ChatManager, createBusService()];
+    },
+});
+
+QUnit.test('messaging menu does not display the Support channel if not available', function (assert) {
+    // the Support channel should only be displayed if a support_token and a support_origin are
+    // specified in the session, which is not the case for this test
+     assert.expect(1);
```
Comment (ged-odoo) [inline addons/im_support/static/tests/systray_no_support_tests.js]: indenting here

File: addons/im_support/static/tests/systray_tests.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
+odoo.define('im_support.systray_tests', function (require) {
+"use strict";
+
+var supportBus = require('im_support.SupportBus');
+var imSupportTestUtils = require('im_support.test_utils');
+
```
Comment (ged-odoo) [inline addons/im_support/static/tests/systray_tests.js]: empty line?

File: addons/im_support/static/tests/systray_tests.js [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            if (route.split('/')[1] === 'odoo_im_support') {
+                assert.step('cors: ' + route);
+            }
+            return this._super.apply(this, arguments);
+        },
+        services: this.services,
+        session: this.supportParams,
+    });
+    testUtils.intercept(messagingMenu, 'call_service', function (ev) {
+        if (ev.data.service === 'local_storage') {
```
Comment (ged-odoo) [inline addons/im_support/static/tests/systray_tests.js]: i see that this is an example of something that should be easily done in a test util function.

File: addons/im_support/static/tests/systray_tests.js [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    var notification = [[false, 'mail.channel'], data];
+    supportBus.trigger('notification', [notification]);
+
+    assert.strictEqual($('.o_chat_window .o_thread_message').length, 1,
+        "there should be a new message in the thread");
+    assert.strictEqual($('.o_chat_window .o_thread_message .o_thread_author ').text().trim(),
+        'An operator', "should correctly display the author");
+    assert.strictEqual($('.o_chat_window .o_thread_message .o_thread_message_avatar').attr('src'),
+        '/mail/static/src/img/odoo_o.png', "should have correct avatar");
+
```
Comment (alexkuhn) [inline addons/im_support/static/tests/systray_tests.js]: check message body is correct

**Pull request closed without merging**